src/lib.rs

```diff
@@ -58,19 +58,66 @@ use std::thread;
 use std::time::Duration;
 
 use crc16::*;
-use rand::seq::sample_iter;
-use rand::thread_rng;
+use rand::{
+    seq::{IteratorRandom, SliceRandom},
+    thread_rng,
+};
 use redis::{
-    Cmd, ConnectionAddr, ConnectionInfo, ErrorKind, IntoConnectionInfo, RedisError, Value,
+    cmd, Cmd, ConnectionAddr, ConnectionInfo, ErrorKind, IntoConnectionInfo, RedisError, Value,
 };
 
 pub use redis::{pipe, Commands, ConnectionLike, PipelineCommands, RedisResult};
 
 const SLOT_SIZE: usize = 16384;
 
+/// This is a Builder of Redis cluster client.
+pub struct Builder<T: IntoConnectionInfo> {
+    initial_nodes: Vec<T>,
+    readonly: bool,
+    password: Option<String>,
+}
+
+impl<T: IntoConnectionInfo> Builder<T> {
+    /// Generate the base configuration for new Client.
+    pub fn new(initial_nodes: Vec<T>) -> Builder<T> {
+        Builder {
+            initial_nodes: initial_nodes,
+            readonly: false,
+            password: None,
+        }
+    }
+
+    /// Connect to a redis cluster server and return a cluster client.
+    /// This does not actually open a connection yet but it performs some basic checks on the URL.
+    /// The password of initial nodes must be the same all.
+    ///
+    /// # Errors
+    ///
+    /// If it is failed to parse initial_nodes or the initial nodes has different password, an error is returned.
+    pub fn open(self) -> RedisResult<Client> {
+        Client::open_internal(self)
+    }
+
+    /// Set password for new Client.
+    pub fn password(mut self, password: String) -> Builder<T> {
+        self.password = Some(password);
+        return self;
+    }
+
+    /// Set read only mode for new Client.
+    /// Default is not read only mode.
+    /// When it is set to readonly mode, all query use replica nodes except there are no replica nodes.
+    /// If there are no replica nodes, it use master node.
+    pub fn readonly(mut self) -> Builder<T> {
```

Comment:
I would include the readonly parameter in the call. For example, I may want to do:

```
builder.readonly(false);
```

Reply:
Okay. Actually I was thinking about whether to have arguments or not.